src/node/test/channels.cpp

```diff
@@ -254,24 +368,115 @@ TEST_CASE("Client/Server key exchange")
   }
 }
 
+TEST_CASE("Switch to endorsed certificates")
```

Comment:
I don't understand the use case for switching to endorsed certificate. The channel shouldn't be established until the node certificates of both peers have been endorsed. What am I missing?

Reply:
Possible. I don't understand the logic for when it's okay to accept self-signed certificates. During the joining process, nodes seem to want to communicate with self-signed certificates until they are properly joined. Are you saying that they shouldn't do that?